test/autorest/generated/paginggroup/pollers.go

```diff
@@ -120,13 +123,26 @@ func (p *productResultPoller) pollUntilDone(ctx context.Context, frequency time.
 	return p.FinalResponse(ctx)
 }
 
-func (p *productResultPoller) handleResponse(resp *azcore.Response) (*ProductResultResponse, error) {
-	result := ProductResultResponse{RawResponse: resp.Response}
-	if resp.HasStatusCode(http.StatusNoContent) {
-		return &result, nil
-	}
-	if !resp.HasStatusCode(pollingCodes[:]...) {
-		return nil, p.pt.handleError(resp)
+func (p *productResultPagerPoller) handleResponse(resp *azcore.Response) (ProductResultPager, error) {
+	u, err := url.Parse(p.pt.finalGetURL())
```

Comment:
This means that we end up performing a GET on the first page URL twice.  What if we added an `*azcore.Response` field to the `productResultPager` that smuggles in `resp` and the first time calling `NextPage()` we use this result instead of performing the GET?

Reply:
done